cfme/middleware/__init__.py

```diff
@@ -53,3 +68,63 @@ def download(extension):
         download_btn("Download as {}".format(extensions_mapping[extension]))
     except:
         raise ValueError("Unknown extention. check the extentions_mapping")
+
+
+def add_deployment(filename, runtime_name=None, enable_deploy=True, cancel=False):
```

Comment:
I think I wrongly explained.
Deployments can be done from different pages of Middleware provider, now from Summary, list of deployments, ..., this is toolbar button, I do not want to limit it in Deployment's class.
But I think it is Middleware specific and no need to have in mixins.